tfx/src/main/scala/com/spotify/noether/tfx/TfmaConverter.scala

```diff
@@ -29,35 +30,69 @@ trait TfmaConverter[A, B, T <: Aggregator[A, B, _]] {
-        underlying
-          .andThenPresent { matrix =>
-            MetricsForSlice
-              .newBuilder()
-              .setSliceKey(SliceKey.getDefaultInstance)
-              .putMetrics(
-                "Noether_BinaryConfusionMatrix",
-                MetricValue
-                  .newBuilder()
-                  .setConfusionMatrixAtThresholds(
-                    newBuilder()
-                      .setMatrices(0,
-                                   ConfusionMatrixAtThreshold
-                                     .newBuilder()
-                                     .setThreshold(underlying.threshold)
-                                     //                  .setFalseNegatives(matrix)))
-                                     .build())
-                      .build())
-                  .build()
-              )
-              .build()
-          }
+  private def denseMatrixToMetric(matrix: DenseMatrix[Long], threshold: Option[Double] = None)
+  : MetricsForSlice = {
+    val tp = matrix.valueAt(1, 1).toDouble
+    val tn = matrix.valueAt(0,0).toDouble
+    val fp = matrix.valueAt(1, 0).toDouble
+    val fn = matrix.valueAt(0, 1).toDouble
+
+    val cmBuilder = ConfusionMatrixAtThreshold.newBuilder()
+      .setFalseNegatives(fn)
+      .setFalsePositives(fp)
+      .setTrueNegatives(tn)
+      .setTruePositives(tp)
+      .setPrecision(tp / (tp + fp))
+      .setRecall(tp / (tp + fn))
+
+    threshold.foreach(cmBuilder.setThreshold(_))
+
+    MetricsForSlice.newBuilder()
+      .setSliceKey(SliceKey.getDefaultInstance)
+      .putMetrics("Noether_ConfusionMatrix",
+        MetricValue.newBuilder()
+          .setConfusionMatrixAtThresholds(ConfusionMatrixAtThresholds.newBuilder()
+            .addMatrices(cmBuilder.build())
+            .build())
+          .build())
+      .build()
+  }
+
+  implicit val errorRateSummaryConverter
+  : TfmaConverter[Prediction[Int, List[Double]], (Double, Long), ErrorRateSummary.type] =
+    new TfmaConverter[Prediction[Int, List[Double]], (Double, Long), ErrorRateSummary.type] {
+      override def convertToTfmaProto(underlying: ErrorRateSummary.type)
+      : Aggregator[Prediction[Int, List[Double]], (Double, Long), MetricsForSlice] =
+        ErrorRateSummary.andThenPresent { ers =>
+          MetricsForSlice.newBuilder()
+            .setSliceKey(SliceKey.getDefaultInstance)
+            .putMetrics("Noether_ErrorRateSummary",
+              MetricValue.newBuilder()
+                .setBoundedValue(BoundedValue.newBuilder()
+                  .setValue(DoubleValue.newBuilder().setValue(ers).build()
+                  ).build())
+                .build()
+            ).build()
+        }
     }
 
+  implicit val binaryConfusionMatrixConverter
+  : TfmaConverter[BinaryPred, Map[(Int, Int), Long], BinaryConfusionMatrix] =
+    (underlying: BinaryConfusionMatrix) => underlying
+      .andThenPresent {
+        cm => denseMatrixToMetric(cm, Some(underlying.threshold))
```

Comment:
Wonder if it's a bit cleaner to define the helper method as 
`private def denseMatrixToMetric(threshold: Option[Double] = None)(cm: DenseMatrix[Long])` so that here we can just do `.andThenPresent(denseMatrixToMetric(Some(underlying.threshold)))`? It's only marginally nicer due to the currying but wdyt?

Reply:
I'll leave this one up to you